symfony/plugins/orangehrmTimePlugin/Dao/ProjectDao.php

```diff
@@ -212,4 +215,90 @@ public function hasTimesheetItemsForProject(int $projectId): bool
+    }
+
+    /**
+     * @param ProjectReportSearchFilterParams $projectReportSearchFilterParams
+     * @return int
+     */
+    public function getProjectReportCriteriaListCount(ProjectReportSearchFilterParams $projectReportSearchFilterParams
+    ): int {
+        return $this->getProjectReportCriteria($projectReportSearchFilterParams)->count();
+    }
+
+    /**
+     * @param ProjectReportSearchFilterParams $projectReportSearchFilterParams
+     * @return Paginator
+     */
+    private function getProjectReportCriteria(ProjectReportSearchFilterParams $projectReportSearchFilterParams
+    ): Paginator {
+        $q = $this->createQueryBuilder(TimesheetItem::class, 'timesheetItem');
+        $q->select(
+            'projectActivity.id AS activityId, projectActivity.name, projectActivity.deleted AS deleted, SUM(COALESCE(timesheetItem.duration, 0)) AS totalDuration'
+        );
+        $q->leftJoin('timesheetItem.projectActivity', 'projectActivity');
+        $q->leftJoin('timesheetItem.timesheet', 'timesheet');
+        $this->setSortingAndPaginationParams($q, $projectReportSearchFilterParams);
+
+        $this->getCommonQuery($projectReportSearchFilterParams, $q);
+        $q->groupBy('projectActivity.id');
+
+        return $this->getPaginator($q);
+    }
+
+    /**
+     * @param ProjectReportSearchFilterParams $projectReportSearchFilterParams
+     * @return string
+     */
+    public function getTotalDurationForProjectReport(ProjectReportSearchFilterParams $projectReportSearchFilterParams): string
+    {
+        $q = $this->createQueryBuilder(TimesheetItem::class, 'timesheetItem');
+        $q->select('SUM(COALESCE(timesheetItem.duration, 0)) AS totalDuration');
+        $q->leftJoin('timesheetItem.projectActivity', 'projectActivity');
+        $q->leftJoin('timesheetItem.timesheet', 'timesheet');
+
+        $this->getCommonQuery($projectReportSearchFilterParams, $q);
+
+        return $q->getQuery()->getSingleScalarResult();
+    }
+
+    /**
+     * @param ProjectReportSearchFilterParams $projectReportSearchFilterParams
+     * @param QueryBuilder $q
+     * @return void
+     */
+    private function getCommonQuery(
+        ProjectReportSearchFilterParams $projectReportSearchFilterParams,
+        QueryBuilder $q
+    ): void {
+        if (!is_null($projectReportSearchFilterParams->getProjectId())) {
+            $q->andWhere('timesheetItem.project = :projectId');
+            $q->setParameter('projectId', $projectReportSearchFilterParams->getProjectId());
+        }
+
+        // TODO BA
+        if (!is_null($projectReportSearchFilterParams->getFromDate()) && !is_null($projectReportSearchFilterParams->getToDate()))
+        {
+            $q->andWhere($q->expr()->between('timesheetItem.date', ':fromDate', ':toDate'))
+                ->setParameter('fromDate', $projectReportSearchFilterParams->getFromDate())
+                ->setParameter('toDate', $projectReportSearchFilterParams->getToDate());
+        }
```

Comment:
try out this by writing some unit tests. I don't think if we added this, `fromDate` and `toDate` interrelated. You don't need to define default date for `fromDate` and `toDate`
```suggestion
        } elseif (!is_null($projectReportSearchFilterParams->getFromDate())) {
            $q->andWhere($q->expr()->gte('timesheetItem.date', ':fromDate'))
                ->setParameter('fromDate', $projectReportSearchFilterParams->getFromDate());
        } elseif (!is_null($projectReportSearchFilterParams->getToDate())) {
            $q->andWhere($q->expr()->lte('timesheetItem.date', ':toDate'))
                ->setParameter('toDate', $projectReportSearchFilterParams->getToDate());
        }
```

Reply:
done